manifests/qpid.pp

```diff
@@ -18,12 +18,39 @@
     require   => Service['qpidd'],
     logoutput => true,
   } ~>
-  exec { 'bind katello entitlements queue to qpid exchange messages that deal with entitlements':
-    command   => "qpid-config --ssl-certificate ${client_cert} --ssl-key ${client_key} -b 'amqps://localhost:5671' bind event ${candlepin_event_queue} '*.*'",
+  exec { 'bind katello entitlements queue to qpid exchange messages that deal with entitlement creation':
+    command   => "qpid-config --ssl-certificate ${client_cert} --ssl-key ${client_key} -b 'amqps://localhost:5671' bind event ${candlepin_event_queue} entitlement.created",
+    onlyif    => "qpid-config --ssl-certificate ${client_cert} --ssl-key ${client_key} -b 'amqps://localhost:5671' queues ${candlepin_event_queue}",
```

Comment:
Dont know how to write it in puppet, but prior this update, one shall see:

    # qpid-config --ssl-certificate=/etc/pki/katello/qpid_client_striped.crt -b "amqps://localhost:5671" exchanges event -r
    Exchange 'event' (topic)
        bind [*.*] => katello_event_queue
    #

while after the change:

    # qpid-config --ssl-certificate=/etc/pki/katello/qpid_client_striped.crt -b "amqps://localhost:5671" exchanges event -r
    Exchange 'event' (topic)
        bind [compliance.created] => katello_event_queue
        bind [entitlement.created] => katello_event_queue
        bind [entitlement.deleted] => katello_event_queue
        bind [pool.created] => katello_event_queue
        bind [pool.deleted] => katello_event_queue
    #

There is not one-line check if one particular binding exists, I am afraid. Just use grep together with above.